src/main/scala/firrtl/annotations/JsonProtocol.scala

```diff
@@ -66,19 +54,30 @@ object JsonProtocol {
   def deserialize(in: JsonInput): Seq[Annotation] = deserializeTry(in).get
 
   def deserializeTry(in: JsonInput): Try[Seq[Annotation]] = Try({
-    def throwError() = throw new InvalidAnnotationFileException(prettifyJsonInput(in))
     val parsed = parse(in)
     val annos = parsed match {
       case JArray(objs) => objs
-      case _ => throwError()
+      case _ => throw new InvalidAnnotationJSONException("Unknown Error")
```

Comment:
Might it provide a clue to the problem if you'd use a var here and then add it to the Exception.
`case x => throw new InvalidAnnotationJSONException("Unexpected value in parsing '$x'")